lang/cs/Org.Apache.REEF.Network/Elastic/Driver/Default/DefaultElasticContext.cs

```diff
@@ -0,0 +1,428 @@
+        {
+            var task = value.Id;
+            _nameServer.Unregister(task);
+        }
+
+        /// <summary>
+        /// Used to react when a timeout event is triggered.
+        /// It gets a failed task as input and in response it produces zero or more failure events.
+        /// </summary>
+        /// <param name="alarm">The alarm triggering the timeput</param>
+        /// <param name="msgs">A list of messages encoding how remote Tasks need to reach</param>
+        /// <param name="nextTimeouts">The next timeouts to be scheduled</param>
+        public void OnTimeout(Alarm alarm, ref List<IElasticDriverMessage> msgs, ref List<ITimeout> nextTimeouts)
+        {
+        }
+
+        /// <summary>
+        /// When a new failure state is reached, this method is used to dispatch
+        /// such event to the proper failure mitigation logic.
+        /// It gets a failure event as input and produces zero or more failure response messages
+        /// for tasks (appended into the event).
+        /// </summary>
+        /// <param name="event">The failure event to react upon</param>
+        public void EventDispatcher(ref IFailureEvent @event)
+        {
+            switch ((DefaultFailureStateEvents)@event.FailureEvent)
+            {
+                case DefaultFailureStateEvents.Reconfigure:
+                    var rec = @event as ReconfigureEvent;
+                    OnReconfigure(ref rec);
+                    break;
+
+                case DefaultFailureStateEvents.Reschedule:
+                    var res = @event as RescheduleEvent;
+                    OnReschedule(ref res);
+                    break;
+
+                case DefaultFailureStateEvents.Stop:
+                    var stp = @event as StopEvent;
+                    OnStop(ref stp);
+                    break;
+
+                default:
+                    OnFail();
+                    break;
+            }
+        }
+
+        #endregion Failure Response
+
+        #region Default Failure event Response
+
+        /// <summary>
+        /// Mechanism to execute when a reconfigure event is triggered.
+        /// <paramref name="reconfigureEvent"/>
+        /// </summary>
+        public void OnReconfigure(ref ReconfigureEvent reconfigureEvent)
+        {
+            lock (_statusLock)
+            {
+                _failureStatus = _failureStatus.Merge(
+                    new DefaultFailureState((int)DefaultFailureStates.ContinueAndReconfigure));
+            }
+        }
+
+        /// <summary>
+        /// Mechanism to execute when a reschedule event is triggered.
+        /// <paramref name="rescheduleEvent"/>
+        /// </summary>
+        public void OnReschedule(ref RescheduleEvent rescheduleEvent)
+        {
+            lock (_statusLock)
+            {
+                _failureStatus = _failureStatus.Merge(
+                    new DefaultFailureState((int)DefaultFailureStates.ContinueAndReschedule));
```

Comment:
also, if we are OK with mutable the `FailureState` property (and it is currently fully read and write accessible from the outside - maybe we should tighten it up a bit?), we can write e.g.
```C#
_failureStatus.Update(DefaultFailureState.ContinueAndReschedule);
```

Reply:
Are you proposing to implement a default behavior for `Merge` when mutating? I like it!